app/src/main/java/com/wizeline/heroes/viewmodels/MarvelViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package com.wizeline.heroes.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.wizeline.heroes.Characters
+import com.wizeline.heroes.interfaces.IRepository
+import com.wizeline.heroes.models.Character
+import com.wizeline.heroes.models.MarvelViewState
+import com.wizeline.heroes.toMD5
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class MarvelViewModel: ViewModel() {
+
+    /** WIZELINE KEYS **/
+    /*private val privateKey = "244be7c2496cbf6d331145cc489b4892457cc2c0"
+    private val apikey = "6ffcf49b680b7250a6983acd33731f55"*/
+
+    /** PERSONAL KEYS **/
+    private val privateKey = "e6b903bda99ed895cd906b8f881e59388c611219"
+    private val apikey = "1ace567a5fc6ae56303e8e340df61a16"
+    private val ts = System.currentTimeMillis().toString()
+    private val compositeDisposable = CompositeDisposable()
+
+    private var _marvelViewState = MutableLiveData(MarvelViewState(arrayListOf(), "", false))
+
+    val marvelViewState get() = _marvelViewState
+
+    fun getCharacters(repository: IRepository, offset: Int){
+
+        val hash = (ts + privateKey + apikey).toMD5()
+
+        // Use disposable to prevent observable memory leaks
+        compositeDisposable.add(repository.getCharacters(ts, apikey, hash, offset).observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(Schedulers.io())
+            .subscribe({
+                val currentViewState = marvelViewState.value!!
+                val marvelCharacterList = ArrayList<Character>()
+                it.data.results.forEach {
+                    marvelCharacterList.add(it)
+                }
+                currentViewState.characterList = marvelCharacterList
+                currentViewState.isListUpdated = true
+                _marvelViewState.value = currentViewState
```

Comment:
I think most of this boilerplate updates can be replaced by creating an instance directly and remove any `currentViewState` variable. Since `marvelViewState` should be always the current state to render, there's no need for an additional variable. For example we could do it like this:
```
_marvelViewState.value = MarvelViewState(
    characterList = it.data.results,
    isListUpdated = true
)
```
Same with other references to update `marvelViewState.value`. We just pass around the required data through the instance constructor. Also, that way you will avoid the use of `!!` operator.

Reply:
Done.